OracleSOASuite/dockerfiles/12.2.1.3/Dockerfile.patch

```diff
@@ -0,0 +1,81 @@
+#
```

Comment:
In our projects the patch is a separate sample with its own README for example https://github.com/oracle/docker-images/tree/master/OracleFMWInfrastructure/samples/12213-patch

Reply:
My understanding is if we use the Docker files as in the patch sample it will create additional layers and will cause large image size. This is similar to the update option in the image tool. 

We agreed to create new images with original shiphomes and patches combined to optimize the size. The sample Docker file that we have created Dockerfile.patch achieves that purpose. 

Also keeping in a separate folder will be maintenance headache as we would like to ensure the Dockerfile and Dockerfile.patch should be exactly same except additional step fo applying the patch.

Hence, we decided to keep in the same folder. 

Let me know if you have any disagreement with the above plan.

Reply:
Regardless of providing instructions in our K8s projects with instructions and calling it our prefered method of patching it is a requirement that all our images created with Dockerfiles in oracle/docker-imagtes must have a sample which provides instructions how to patch manually in GitHub using Dockerfiles.

Reply:
All our projects should have similar organization otherwise customers fo to the FMW Infra it is one thing and then SOA and it is another.

Reply:
As I mentioned in my other update, Docker image creation on patches we followed the following best practices based on your recommendation -

1) Create the image instead of update. The patch sample that is present in WLS is actually updating the release image with patches. The patch sample that we have written will recreate the image with installers and patches

2) The above create option helps upper stack products to optimize the image sizes otherwise it increases to the size of 10GB or more.

Agree all projects should have similar organization but we need to do something differently as Upper stack products are more bigger in size and complicated. 

Let me know if we can have an offline discussions.

Reply:
I agree with this approach.  Is it documented in the README how to patch?

Reply:
Yes, image creation with patch is documented in the README [link](https://github.com/sbattagi/docker-images/blob/master/OracleSOASuite/dockerfiles/README.md#building-docker-image-with-bundle-patches-for-soa).

Reply:
On further internal discussions, we plan to make Dockerfile.patch as the only Dockerfile. This way we can have only one file and if there are any patches to be applied the customers will need to copy the patches into the predefined location of 12.2.1.3/patches. If any zip file present the image will be created with patches in it. If the folder is empty then image will be created with any patches. Pls review and let us know if this sounds okay.

Reply:
That sounds fine to me, btw.

Reply:
I am fine with this approach as well, please make sure that it is documented in the README.md and you provide an example.

Reply:
I agree with this approach

Reply:
Checked in the changes for this agreed approach (single Dockerfile for image build with and without patches). Updated the documentation as well.